### ISSUE TITLE
--verifytypes incorrectly marks ambiguous base override with properties

### ISSUE THREAD

**Issue by Rapptz** [microsoft/pyright#3293]
**Describe the bug**

Using `pyright --verifytypes package` causes an error

```
 error: Ambiguous base class override
    Type declared in base class is "property"
    Type inferred in child class is "int"
```

**To Reproduce**

Create a package with a `py.typed` file and an  `__init__.py` file with this content:

```py
class Foo:
    def __init__(self):
        self._foo: int = 20

    @property
    def foo(self) -> int:
        return self._foo

    @foo.setter
    def foo(self, value: int) -> None:
        self._foo = value


class Bar(Foo):

    def __init__(self) -> None:
        super().__init__()
        self.foo = 23
```

and run `pyright --verifytypes package`

**Expected behavior**

No error.

**VS Code extension or command-line**
Command line.
```
$ pyright --version
pyright 1.1.234
```

**Additional context**

This is calling the setter of the property at runtime, so the code should be correct and there shouldn't be any incompatibilities  or ambiguities here to my knowledge.

I spent some time trying to bisect and this used to error in versions before <1.1.221 due to a missing type annotation. It started to not error in `1.1.221` up until `1.1.225` where it started erroring again due to a [documented behavioural change](https://github.com/microsoft/pyright/releases/tag/1.1.225).

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.235, which I just pblished.
